addons/pos_restaurant/static/src/app/models/restaurant_table.js

```diff
@@ -85,5 +85,8 @@ export class RestaurantTable extends Base {
     getName() {
         return this.table_number.toString();
     }
+    get children() {
+        return this.models["restaurant.table"].filter((t) => t.parent_id?.id === this.id);
```

Comment:
```suggestion
        return this["<-restaurant.table.parent_id"];
```

I tested and this suggestion seems to be working.